# Results with Merged Dataset
Remember to use your new dataset, `combined_dataset.csv`

In [1]:
# load dataset
import pandas as pd

df = pd.read_csv('combined_dataset.csv')

In [2]:
df.head(1)

model_2008  displ_2008  cyl_2008 trans_2008 drive_2008 fuel_2008  \
0  ACURA RDX         2.3         4    Auto-S5        4WD  Gasoline   

  veh_class_2008  air_pollut_2008  city_mpg_2008  hwy_mpg_2008   ...     \
0            SUV              7.0           17.0          22.0   ...      

        trans  drive      fuel  veh_class  air_pollution_score  city_mpg  \
0  SemiAuto-6    2WD  Gasoline  small SUV                  3.0      20.0   

  hwy_mpg cmb_mpg greenhouse_gas_score smartway  
0    28.0    23.0                    5       No  

[1 rows x 26 columns]

### Q5: For all of the models that were produced in 2008 that are still being produced now, how much has the mpg improved and which vehicle improved the most?

In [82]:
# Create a new dataframe, model_mpg, that contain the mean combined mpg values in 2008 and 2018 for each unique model

# To do this, group by model and find the mean cmb_mpg_2008 and mean cmb_mpg for each.
model_mpg_08 = df.groupby('model', as_index=False).cmb_mpg_2008.mean()
model_mpg_18 = df.groupby('model', as_index=False).cmb_mpg.mean()

model_mpg = model_mpg_08.merge(model_mpg_18, left_on='model', right_on='model', how='inner')

In [83]:
model_mpg.head()

model  cmb_mpg_2008    cmb_mpg
0  ACURA RDX     19.000000  22.500000
1    AUDI A3     23.333333  28.000000
2    AUDI A4     21.000000  27.000000
3    AUDI A6     19.666667  25.666667
4  AUDI A8 L     16.500000  22.000000

In [86]:
# Create a new column, mpg_change, with the change in mpg
# Subtract the mean mpg in 2008 from that in 2018 to get the change in mpg
model_mpg['mpg_change'] = model_mpg.cmb_mpg - model_mpg.cmb_mpg_2008

In [88]:
model_mpg.head(20)

model  cmb_mpg_2008    cmb_mpg  mpg_change
0                ACURA RDX     19.000000  22.500000    3.500000
1                  AUDI A3     23.333333  28.000000    4.666667
2                  AUDI A4     21.000000  27.000000    6.000000
3                  AUDI A6     19.666667  25.666667    6.000000
4                AUDI A8 L     16.500000  22.000000    5.500000
5                  AUDI Q7     15.000000  21.000000    6.000000
6                  AUDI S4     15.500000  24.000000    8.500000
7                  AUDI S5     16.000000  24.000000    8.000000
8            AUDI TT Coupe     21.666667  26.000000    4.333333
9         AUDI TT Roadster     21.333333  26.000000    4.666667
10                BMW 750i     20.000000  19.500000   -0.500000
11           BMW Alpina B7     18.000000  19.000000    1.000000
12      BMW M6 Convertible     14.000000  16.500000    2.500000
13            CADILLAC CTS     20.000000  22.800000    2.800000
14       CADILLAC Escalade     14.000000  17.000000    3.000000
15      CHEVROLET Corvette     18.000000  17.750000   -0.250000
16       CHEVROLET Equinox     19.000000  27.833333    8.833333
17        CHEVROLET Impala     20.500000  21.250000    0.750000
18        CHEVROLET Malibu     22.333333  33.000000   10.666667
19  CHEVROLET Silverado 15     16.000000  17.312500    1.312500

In [89]:
# Find the vehicle that improved the most
# Find the max mpg change, and then use query or indexing to see what model it is!
model_mpg.mpg_change.max()

16.533333333333339

In [90]:
model_mpg.query('mpg_change == "16.533333333333339"')

model  cmb_mpg_2008  cmb_mpg  mpg_change
71  VOLVO XC 90     15.666667     32.2   16.533333

VOLVO XC 90 improved the most in mileage, increasing its mileage by 16.5 miles.